chroma-agent/chroma_agent/device_plugins/linux_components/block_devices.py

```diff
@@ -134,149 +224,25 @@ def composite_device_list(self, source_devices):
-                return None
-
-        block_device_nodes = {}
-        node_block_devices = {}
-
-        def parse_block_dir(dev_dir, parent = None):
-            """ Parse a dir like /sys/block/sda (must contain 'dev' and 'size') """
-            size = 0
-            device_name = dev_dir.split(os.sep)[-1]
-
-            try:
-                major_minor = open(os.path.join(dev_dir, "dev")).read().strip()
-                size = int(open(os.path.join(dev_dir, "size")).read().strip()) * 512
-            except IOError:
-                pass
-
-            # Exclude zero-sized devices
-            if size == 0:
-                return
-
-            # Exclude ramdisks, floppy drives, obvious cdroms
-            if re.search("^ram\d+$", device_name) or\
-               re.search("^fd\d+$", device_name) or\
-               re.search("^sr\d+$", device_name):
-                return
-
-            # Exclude read-only devices and removed media or devices
-            try:
-                # Never use 'w' in the built-in open() or it'll create a 0 length file where a
-                # device was removed!
-                fd = os.open("/dev/%s" % device_name, os.O_WRONLY)
-            except OSError, e:
-                # EROFS: Device is read-only
-                # ENOENT: No such file or directory
-                # NO_MEDIA_ERRNO: No medium found
-                if e.errno in [errno.EROFS, errno.ENOENT, errno.NO_MEDIA_ERRNO]:
-                    return
-            else:
-                os.close(fd)
-
-            # Resolve a major:minor to a /dev/foo
-            path = get_path(major_minor, device_name)
-            if path:
-                if parent:
-                    partition_number = int(re.search("(\d+)$", device_name).group(1))
-                else:
-                    partition_number = None
-
-                block_device_nodes[major_minor] = self._device_node(major_minor, path, size, parent, partition_number)
-                node_block_devices[path] = major_minor
-
-            return major_minor
-
-        for dev_dir in glob.glob("/sys/block/*"):
-            major_minor = parse_block_dir(dev_dir)
-
-            partitions = glob.glob(os.path.join(dev_dir, "*/dev"))
-            for p in partitions:
-                parse_block_dir(os.path.split(p)[0], parent = major_minor)
-
-        # Finally create the normalized maps for /dev to /dev/disk/by-path & /dev/disk/by-id
-        # and then /dev/disk/by-path & /dev/disk/by-id to /dev/mapper
-        ndp.add_normalized_list(dev_nodes, by_path_nodes)
-        ndp.add_normalized_list(dev_nodes, by_id_nodes)
-        ndp.add_normalized_list(by_path_nodes, mapper_devs)
-        ndp.add_normalized_list(by_id_nodes, mapper_devs)
-
-        return block_device_nodes, node_block_devices
+        def build_paths(x):
+            out = []
 
+            for path in x['paths']:
```

Comment:
return [(x['major_minor'], path) for path in x['paths'] if path.startswith(folder)]

Reply:
Done.